### ISSUE TITLE
Integrate Fuzzy Search for suggestions

### ISSUE THREAD

**Issue by ad1992** [react-tags/react-tags#984]
Let's use [fuzzify](https://github.com/ad1992/fuzzify) - dogfooding 📦

**Comment by shrilakshmishastry** [comment]
Hey @ad1992 I would like to take up this task

**Comment by ad1992** [comment]
sure go ahead @shrilakshmishastry

**Comment by shrilakshmishastry** [comment]
Hey @ad1992 here is the short description of the above task 

**Problem statement**

- Use fuzzy logic to render suggestions in react-tags based on query input

**current functionality**

The method used to filter suggestions is filteredSuggestions(). This method uses string matching for all items in the suggestion. Keeping exact match in the beginning and partial match at the end

**new functionality**

replace this filtering methodology with fuzzy logic.

expected from the task is to use a library called [fuzzify](https://github.com/ad1992/fuzzify)

**Comment by shrilakshmishastry** [comment]
is above understanding is correct?

**Comment by ad1992** [comment]
> is above understanding is correct?

@shrilakshmishastry yes, but to clarify - you need not implement your own fuzzy logic for rendering the suggestions as that's taken care of by fuzzify, you need to only integrate the library [fuzzify](https://github.com/ad1992/fuzzify) npm package in this library for suggestions. 
Let me know if you have any questions

**Comment by shrilakshmishastry** [comment]
It's clear to me.

**Comment by ad1992** [comment]
@shrilakshmishastry if you face any issues in integration, please let me know as the [fuzzify](https://github.com/ad1992/fuzzify) is at a very early stage and I am looking for feedback as well :)

**Comment by AndyClausen** [comment]
Wouldn't it be better to expose an `onSearch` event to allow users to provide their own search functionality? Then you can do the search on backend or database level, for example. You can still provide a default search. This way, no more deps are added as well.
